mongo/cursor.go

```diff
@@ -67,6 +68,45 @@ func newEmptyCursor() *Cursor {
 	return &Cursor{bc: driver.NewEmptyBatchCursor()}
 }
 
+// NewMockCursor creates a new Cursor pre-loaded with the provided documents, error and registry. If no registry is provided,
```

Comment:
Nit: Though the only known use case is mocking. I suggest avoiding `Mock` in the API. Perhaps this could be named `NewCursorFromDocuments` or similar?

My concern is that we may later have use case of creating a cursor from a slice of documents, and using `NewMockCursor` seems odd.

Reply:
Sounds good. Changed to `NewCursorFromDocuments`.